app/javascript/controllers/federation_controller.js

```diff
@@ -0,0 +1,46 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="federation"
+export default class extends Controller {
+    static values = {
+        config: Object,
+      }
+    static targets = ['chips']
+    connect() {
+      this.#initIconsStyle()
+      this.#initInputChips()
+    }
+    #initIconsStyle(){
+      const style = document.createElement('style');
+      let styles = '';
+      Object.entries(this.configValue).forEach(([key, portal]) => {
+          styles += `.federated-icon-${key} path { fill: ${portal.color} !important; }\n`;
+      });
+      style.innerHTML = styles;
+      document.head.appendChild(style);
+    }
+
+    #initInputChips(){
```

Comment:
this requires a loader and to be disabled by default, otherwise, you can click on a portal, and then you see the selected chips afterward disabled.